src/Native/Runtime/arm64/InteropThunksHelpers.S

```diff
@@ -21,7 +21,11 @@ POINTER_SIZE = 0x08
         // Custom calling convention:
         //      xip0 pointer to the current thunk's data block (data contains 2 pointer values: context + target pointers)
 
+#ifdef FEATURE_EMULATED_TLS
+        GETTHUNKDATA_ETLS_9
```

Comment:
This may need to preserve floating point argument registers too. What are the chances that the TLS access will modify them? 

If we believe that we can get away without saving them with fingers crossed, it may be worth a comment at least.

Reply:
Unix AMD64 has the same problem. We are saving floating point registers there.

An alternative way to deal with this problem would be to use a non-standard register to pass the context into the stub. It avoids problems with saving all register on a hot path. It is what CoreCLR does and RyuJIT has support for it.  Switching CoreRT to a scheme like would require some refactoring. It can be done if we wanted to make marshalled delegates on platforms like this. Probably not worth it since we would like to switch everything that matters to `[UnmanagedCallersOnly]` scheme that is faster and much cleaner overall.

Reply:
I checked the source code and had a look in the debugger. The floating point registers are not touched. I added a comment. But if you think it's better to store them I will do it. Based on how much work the TLS emulation need to do to get the value it would not matter that much if we save them

Reply:
Does the TLS emulation have any rare paths - one-time initializations, etc.? They usually cause problems for shortcuts like this.

I am ok with just having a comment for now.

Reply:
The one-time initialization,is done with another function and I could spot anything else that make use of the floating point operations.